beacon-chain/execution/service.go

```diff
@@ -160,10 +162,16 @@ type Service struct {
 func NewService(ctx context.Context, opts ...Option) (*Service, error) {
 	ctx, cancel := context.WithCancel(ctx)
 	_ = cancel // govet fix for lost cancel. Cancel is handled in service.Stop()
-	depositTrie, err := trie.NewTrie(params.BeaconConfig().DepositContractTreeDepth)
-	if err != nil {
-		cancel()
-		return nil, errors.Wrap(err, "could not set up deposit trie")
+	var depositTrie cache.MerkleTree
+	var err error
+	if features.Get().EnableEIP4881 {
+		depositTrie = depositsnapshot.NewDepositTree()
+	} else {
+		depositTrie, err = trie.NewTrie(params.BeaconConfig().DepositContractTreeDepth)
+		if err != nil {
+			cancel()
```

Comment:
Not sure you need to cancel here. There is a not that cancel is handled in service.Stop